app/javascript/reducers/products.js

```diff
@@ -0,0 +1,43 @@
+import { ProductsActions } from '../actions/productsActions.js'
+import update from 'immutability-helper';
+import {normalizeProducts} from '../helpers/JsonNormalizer';
+
+let products_list = default_props && default_props.developer ? normalizeProducts(default_props.developer.products) : {byAddress: {}, allIds: []}
+const initialState = {
+  inProgress: false,
+  registration_steps: {
+    load_to_ipfs: {status: 'waiting', text: 'Loading metadata to IPFS'},
```

Comment:
Is reducer good place to store structure like load_to_ipfs ?

Reply:
I used this for visualisation bot registration process in UI.
Do you propose store in reducer something like this:
```js
registration_steps: {
  load_to_ipfs: 'waiting',
  approve: 'waiting',
  add_bot: 'waiting'
}
```
and `text` and `stepsOrder` move to component where we used these steps?